lib/vector-core/src/event/metric/tags.rs

```diff
@@ -324,62 +351,37 @@ impl MetricTags {
         self.0.get(name).and_then(TagValueSet::as_single)
     }
 
-    pub fn insert(&mut self, name: String, value: TagValue) -> Option<TagValue> {
+    pub fn insert(&mut self, name: String, value: String) -> Option<String> {
         self.0
             .insert(name, TagValueSet::from([value]))
             .and_then(TagValueSet::into_single)
     }
 
-    pub fn remove(&mut self, name: &str) -> Option<TagValue> {
+    pub fn remove(&mut self, name: &str) -> Option<String> {
```

Comment:
Hmm, I realize this is to keep the interfaces the same, but sort of as a forward-looking question: will this eventually return `Option<TagValue>`? So that it can be detected when a bare tag has been removed?

It feels a _little_ weird that we could remove a bare tag here but not actually be told that we did.